src/test/kotlin/racingcar/RacingGameTest.kt

```diff
@@ -0,0 +1,49 @@
+package racingcar
+
+import io.kotest.core.spec.style.ShouldSpec
+import io.kotest.inspectors.forAll
+import io.kotest.matchers.shouldBe
+import racingcar.domain.Car
+import racingcar.domain.RacingGame
+
+class RacingGameTest : ShouldSpec({
+
+    val tryCount = 10
+    val cars: List<Car> = listOf(
+        Car(name = "pobi"),
+        Car(name = "crong"),
+        Car(name = "honux")
+    )
+
+    beforeTest {
+        cars.forAll { it.location = Car.CAR_DEFAULT_LOCATION }
+    }
+
+    context("N대의 자동차를 M번 이동하는 게임이 시작되면") {
+        should("M개의 게임결과가 반환된다") {
+            val racingGameResults = RacingGame(cars, tryCount).run { 1 }
+            racingGameResults.size shouldBe tryCount
+        }
+
+        should("각 게임결과에는 N대의 자동차가 이동한 결과가 저장된다") {
+            val racingGameResults = RacingGame(cars, tryCount).run { 1 }
+            racingGameResults.forAll { gameResult ->
+                gameResult.cars.size shouldBe cars.size
+            }
+        }
+
+        context("게임 횟수마다 자동차가 전진한다면") {
+            should("각 자동차의 위치는 초기 위치로부터 tryCount 만큼 이동한 위치이다") {
+                RacingGame(cars, tryCount).run { (4..9).random() }
+                cars.forAll { it.location shouldBe tryCount }
```

Comment:
또한 조금 번거로울 수 있지만 이렇게 반복문을 돌리는 것보다는 테스트 코드의 검증단은 하드코딩을하는 것도 나쁘지 않다고 생각해요. 😃  

이 코멘트가 조금 납득이 안가실 수 있을 것 같은데, 가급적 테스트 코드에는 로직이 안들어가는 것이 좋습니다. 
직관적으로 이 테스트가 어떤 테스트인지를 한눈에 드러내는 것이 더 좋다고 생각해요. 😉

```suggestion
                cars[0].location shouldBe tryCount
                cars[1].location shouldBe tryCount
                cars[2].location shouldBe tryCount
```

Reply:
이 부분은 공감이되네요!!!! 테스트코드는 반복을 줄이기보다 직관적으로 빠르게 볼 수 있도록 하는 것이 좋은 것 같아요. 저도 회사 프로젝트를 처음 접했을 때, 도메인을 이해하기 위해 테스트코드를 많이 보는 편인데요. 아이러니하게 테스트코드를 이해하는데 또 시간이 드는거 있죠...! 😢 😢

Reply:
저도 테스트 코드로 도메인을 파악하는 방식을 좋아하는데요. 😃
테스트에 로직이 많이 들어가는순간 결국 도메인 로직 읽는거랑 별 차이가 없어지더라구요. 😂